### PR TITLE
mailman: refactor package structure

### LINKED ISSUE #158424: python3.pkgs.mailman is not a Python module
`python3.pkgs.mailman` is not a Python module and `python310.pkgs.mailman` actually uses Python 3.9!???
It can also not be added to `python3Packages` with the correct Python version cause it overrides sqlalchemy.
Thus mailman and the packages depending on it need to be removed from `python-packages.nix`.
cc @mweinelt @jonringer

### PR DESCRIPTION
**Pull request by Ma27**
###### Description of changes

* Removed unused `.package`-option.
* Added explicit postgresql support.
* Create a new meta-package for mailman to make sure each component has
  the **same** python and packages can be downgraded if needed (e.g.
  psycopg2 or sqlalchemy) without interfering with `pythonPackages` in any way.
* Document why certain python overrides are needed.

Closes #170035
Closes #158424

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6548,7 +6540,11 @@ in {
 
   poster3 = callPackage ../development/python-modules/poster3 { };
 
-  postorius = callPackage ../servers/mail/mailman/postorius.nix { };
+  postorius = throw "Please use pkgs.mailmanPackages.postorius";
+  hyperkitty = throw "Please use pkgs.mailmanPackages.hyperkitty";
+  mailman = throw "Please use pkgs.mailman";
+  mailman-hyperkitty = throw "Please use pkgs.mailmanPackages.mailman-hyperkitty";
+  mailman-web = throw "Please use pkgs.mailman-web";
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:6547]: These should go in `python-aliases.nix`.

Comment (Ma27) [reply pkgs/top-level/python-packages.nix]: Done.

**Comment by alyssais** [comment]
Thanks for putting some energy towards mailman!

I don't think we need to share the package set between mailman and mailman-web, because they're independent services that communicate over a REST API.  It's incidental that they're even both written in the same language.  A change I've always meant to make was separating `pythonEnv` into ones for Mailman and malman-web for that reason.  What do you think?

**Comment by dotlambda** [comment]
Can we get this into 22.05?

**Comment by Ma27** [comment]
> Can we get this into 22.05?

Strongly in favor.

> A change I've always meant to make was separating pythonEnv into ones for Mailman and malman-web for that reason. What do you think?

Shouldn't be too hard I guess.... We should still use a custom package set because of some shared things (such as hyperkitty + the psycopg2 fix), but we could add `buildMailmanEnv` and `buildMailmanWebEnv` instead.

Since some conflicts need to be resolved anyways, I can take care of that during the next few days.

**Comment by Ma27** [comment]
Rebased and split the envs into two.

**Comment by Ma27** [comment]
@dotlambda @alyssais anything else to add? %)

**Review by dotlambda — APPROVED** [review]
LGTM, but I didn't test it since I don't run mailman.

**Pull request merged**